# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1225604652 -3600
#      Sun Nov 02 06:44:12 2008 +0100
# Node ID f2e050b6219912307a396a3c056bb1f660a9d8ae
# Parent  f7d44b6a74df4e81acfa47c6b74bed845ffb5bb9
fix dim check in Array<T>::delete_elements

diff --git a/liboctave/Array.cc b/liboctave/Array.cc
--- a/liboctave/Array.cc
+++ b/liboctave/Array.cc
@@ -1347,17 +1347,17 @@ Array<T>::delete_elements (const idx_vec
         }
     }
 }
 
 template <class T>
 void 
 Array<T>::delete_elements (int dim, const idx_vector& i)
 {
-  if (dim > ndims ())
+  if (dim < 0 || dim >= ndims ())
     {
       (*current_liboctave_error_handler)
         ("invalid dimension in delete_elements");
       return;
     }
 
   octave_idx_type n = dimensions (dim);
   if (i.is_colon ())
diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -1,12 +1,14 @@
 2008-11-02  Jaroslav Hajek <highegg@gmail.com>
 
 	* idx-vector.cc (idx_vector::is_complement): Set resulting extent
 	correctly.
+	* Array.cc (Array<T>::delete_elements (int, const idx_vector&)):
+	Fix check for invalid dim.
 
 2008-10-31  Jaroslav Hajek  <highegg@gmail.com>
 
 	* idx-vector.h (idx_vector::idx_range_rep::extent): Don't change
 	extent when len is zero.
 	* idx-vector.h (idx_vector::idx_range_rep::idx_range_rep (void)):
 	Create empty range by default.
 
